index.js

```diff
@@ -353,6 +353,7 @@ RedirectableRequest.prototype._processResponse = function (response) {
     // Evaluate the beforeRedirect callback
     if (typeof this._options.beforeRedirect === "function") {
       try {
+        this._options.responseHeaders = this._currentRequest.res.headers;
```

Comment:
No need to have this in the `try`; also, should not modify the `_options` object

Reply:
I do hope I did understand you correct. See push ;)